api/domains.go

```diff
@@ -9,6 +9,50 @@ import (
 	"time"
 )
 
+// DomainsResponse represents an API response to GET /account/{accountId}/domains
+type DomainsResponse struct {
+	DomainName string `json:"domain_name"`
+	Engaged    bool   `json:"engaged"`
+}
+
+// Domains returns a list of an account's domains
+func Domains(accountID int) (d []DomainsResponse, err error) {
+	u := BaseURL()
+	u.Path += fmt.Sprintf("/account/%d/domains", accountID)
+
+	ctx, cancel := context.WithTimeout(context.Background(), Timeout)
+	defer cancel()
+
+	resp, err := request(ctx, http.MethodGet, u, nil)
+	if err != nil {
+		return d, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		switch resp.StatusCode {
+		case 401:
+			return d, ErrStatusUnauthorized
+		case 403:
+			return d, ErrStatusForbidden
+		default:
+			return d, prettyTxIDError(resp)
+		}
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
`ioutil` is getting deprecated, no?

Reply:
Yep it is 🎉 

Right now I'm aiming for consistency with the rest of the code base. 

I'll put up a PR to upgrade all the legacy packages once 1.16 drops.

Reply:
In this case you don't really need it:

```go
err := json.NewDecoder(resp.Body).Decode(&d)
return d, err
```